crates/nu-std/std/bench/mod.nu

```diff
@@ -63,9 +62,17 @@ Benchmark 2: { 2 ** 4 }
 
 { 2 + 4 } ran
     1 times faster than { 2 ** 4 }"
+@example "use --setup to compile before benchmarking" { bench { ./target/release/foo } --setup { cargo build --release } }
+@example "use --warmup to fill the disk cache before benchmarking" { bench { fd } { jwalk . -k } -w 1 -n 10 }
 export def main [
-    ...codes: closure        # the piece(s) of `nushell` code to measure the performance of
+    ...commands: closure     # the piece(s) of `nushell` code to measure the performance of
     --rounds (-n): int = 50  # the number of benchmark rounds (hopefully the more rounds the less variance)
+    --warmup (-w): int = 0   # the number of warmup rounds (not timed) to do before the benchmark, useful for filling the disk cache in I/O-heavy programs
+    --setup (-s): closure    # command to run before all benchmarks
+    --prepare (-S): closure  # command to run before each benchmark (same as `--setup` if only doing one benchmark)
+    --cleanup (-c): closure  # command to run after all benchmarks
+    --conclude (-C): closure # command to run after each benchmark (same as `--cleanup` if only doing one benchmark)
```

Comment:
To me `--prepare` and `--conclude ` read somewhat ambiguously, if it is run after each iteration (which would allow unmeasured resets between each measured section) or each kind of benchmark (which it does in the current code).